packages/utils/src/optionsList.ts

```diff
@@ -2,18 +2,24 @@ import toConstant from "./toConstant";
 import { RJSFSchema, EnumOptionsType } from "./types";
 
 /** Gets the list of options from the schema. If the schema has an enum list, then those enum values are returned. The
- * labels for the options will be the same as the `value`. If the schema has a `oneOf` or `anyOf`, then the value is the
- * list of `const` values from the schema and the label is either the `schema.title` or the value.
+ * labels for the options will be extracted from the non-standard, RJSF-deprecated `enumNames` if it exists, otherwise
+ * the label will be the same as the `value`. If the schema has a `oneOf` or `anyOf`, then the value is the list of
+ * `const` values from the schema and the label is either the `schema.title` or the value.
  *
  * @param schema - The schema from which to extract the options list
  * @returns - The list of options from the schema
  */
 export default function optionsList(
-  schema: RJSFSchema
+  schema: RJSFSchema & { enumNames?: string[] }
```

Comment:
Please move this inside like the code had done previous to the removal of enumNames

Reply:
```
    // enumNames is not officially on the RJSFSchema, so hack them on here
    const schemaHack: { enumNames?: string[] } = schema as { enumNames?: string[] };
    const schemaHack: { enumNames?: string[] } = schema as {
      enumNames?: string[];
    };
    return schema.enum.map((value, i) => {
      const label = (schemaHack.enumNames && schemaHack.enumNames[i]) || String(value);
      const label =
        (schemaHack.enumNames && schemaHack.enumNames[i]) || String(value);
      return { label, value };
    });
````